consumers/campaigner/main.go

```diff
@@ -243,12 +246,40 @@ func processSubscribers(
 	return nil
 }
 
-func setStatusFailed(ctx context.Context, store storage.Storage, campaign *entities.Campaign) error {
+// overwriting the callback func for max attempts reached to insert into campaign failed logs.
+func (h *MessageHandler) LogFailedMessage(m *nsq.Message) {
+	if m == nil {
+		return
+	}
+	msg := new(entities.CampaignerTopicParams)
+	err := json.Unmarshal(m.Body, &msg)
+	if err != nil {
+		logrus.WithField("body", string(m.Body)).
+			WithError(err).Error("Malformed JSON message.")
+		return
+	}
+	log := &entities.CampaignFailedLog{
+		ID:          ksuid.New(),
+		UserID:      msg.UserID,
+		CampaignID:  msg.CampaignID,
+		Description: "Exceeded max attempts for preparing the campaign",
+		CreatedAt:   time.Time{},
+	}
+	err = storage.CreateCampaignFailedLog(context.Background(), log)
```

Comment:
there is no database in context.Background() you can use the store from MessageHandler (h.store)

Reply:
Also i think we should update the campaign status here as well

Reply:
CampaignFailedLog is doing that in transaction @gudgl